File: Examples/CSharp/Programming-Documents/Document/CheckBoxTypeContentControl.cs

using System.IO;
using Aspose.Words;
using System;
using Aspose.Words.Markup;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Working_With_Document
{
    class CheckBoxTypeContentControl
    {
        public static void Run()
        {
            //ExStart:CheckBoxTypeContentControl
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_WorkingWithDocument();
            //Open the empty document
            Document doc = new Document();

            DocumentBuilder builder = new DocumentBuilder(doc);
            StructuredDocumentTag SdtCheckBox = new StructuredDocumentTag(doc, SdtType.Checkbox, MarkupLevel.Inline);

            //Insert content control into the document
            builder.InsertNode(SdtCheckBox);
            dataDir = dataDir + "CheckBoxTypeContentControl_out_.docx";

            doc.Save(dataDir, SaveFormat.Docx);
            //ExEnd:CheckBoxTypeContentControl
            Console.WriteLine("\nCheckBox type content control created successfully.\nFile saved at " + dataDir);
        }        
    }
}

File: Examples/VisualBasic/Programming-Documents/Document/CheckBoxTypeContentControl.vb
Imports System.IO
Imports Aspose.Words
Imports Aspose.Words.Markup
Class CheckBoxTypeContentControl
    Public Shared Sub Run()
        ' ExStart:CheckBoxTypeContentControl
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_WorkingWithDocument()
        ' Open the empty document
        Dim doc As New Document()

        Dim builder As New DocumentBuilder(doc)
        Dim SdtCheckBox As New StructuredDocumentTag(doc, SdtType.Checkbox, MarkupLevel.Inline)

        ' Insert content control into the document
        builder.InsertNode(SdtCheckBox)
        dataDir = dataDir & Convert.ToString("CheckBoxTypeContentControl_out.docx")

        doc.Save(dataDir, SaveFormat.Docx)
        ' ExEnd:CheckBoxTypeContentControl
        Console.WriteLine(Convert.ToString(vbLf & "CheckBox type content control created successfully." & vbLf & "File saved at ") & dataDir)
    End Sub
End Class

